### PR TITLE
Add assert when the variable temperature_for_viscosity becomes zero.

### PR DESCRIPTION
**Pull request by MFraters**
Prevents division by zero when viscosity_diffusion is computed.

### CODE REVIEW

File: source/material_model/visco_plastic.cc [hunk 1]
```diff
@@ -105,6 +105,14 @@ namespace aspect
       // gradient used when calculating the viscosity. This allows the same activation volume
       // to be used in incompressible and compressible models.
       const double temperature_for_viscosity = temperature + adiabatic_temperature_gradient_for_viscosity*pressure;
+      Assert(temperature_for_viscosity != 0, ExcMessage(
+               "The temperature used in the caluclation of the visco platic rheology is zero. "
```
Comment (tjhei) [inline source/material_model/visco_plastic.cc]: typo "calculation"

**Comment by tjhei** [comment]
Is this for a situation where one incorrectly sets up her problem and you really get precisely 0.0 here? Otherwise a check with ``(abs(x)<eps)`` might be more appropriate?

**Comment by MFraters** [comment]
In the case I encountered this, the temperature and the adiabatic_temperature_gradient_for_viscosity where exactly zero, which led to an SIGFPE which was quite hard to track. So I made this mainly with the intention to prevent other people from wasting time on tracking the floating point exception. I think that is it is not exactly 0.0, there won't be an SIGFPE, although other problems may arise of course. So I am not sure which one is more appropriate.

**Comment by tjhei** [comment]
Well, this is better than nothing and probably not worth worrying about.

**Pull request merged**